include/diijkstra.h

```diff
@@ -0,0 +1,33 @@
+#ifndef DIIJKSTRA_H
+#define DIIJKSTRA_H
+
+#include "frameAndLink.h"
+#include "transmem.h"
+#include <unordered_map>
+
+struct Path;
+
+class Diijkstra
+{
+public:
+    Diijkstra(std::unordered_map<FrameID, Frame*> frameID2Frame)
+    : frameID2Frame(frameID2Frame)
+    {}
+
+    bool calculateShortestPath(Path &path);
```

Comment:
Since I plan to rework the path searching algorithm I will let it be like this. Most likely the whole class will not be needed after rework.